File: students/students/src/MON.SPA/src/models/diploma/diplomaCreateRequestModel.js
import Constants from '@/common/constants.js';

export class DiplomaCreateRequestModel {
  constructor(obj = {}, moment = {}) {
    this.id = obj.id;
    this.personId = obj.personId;
    this.requestingInstitutionId = obj.requestingInstitutionId;
    this.currentInstitutionId = obj.currentInstitutionId;
    this.basicDocumentId = obj.basicDocumentId;
    this.registrationNumber = obj.registrationNumber;
    this.registrationNumberYear = obj.registrationNumberYear;
    this.registrationDate = (obj.registrationDate && moment) ? moment(obj.registrationDate).format(Constants.DATEPICKER_FORMAT) : '';
    this.note = obj.note;
    this.isGranted = obj.isGranted;
    this.arbitraryCurrentInstitutionName = obj.arbitraryCurrentInstitutionName;
  }
}

File: students/students/src/MON.Models/Diploma/DiplomaCreateRequestModel.cs
namespace MON.Models.Diploma
{
    using System;

    public class DiplomaCreateRequestModel
    {
        public int? Id { get; set; }
        public int PersonId { get; set; }
        public int RequestingInstitutionId { get; set; }
        public int? CurrentInstitutionId { get; set; }
        public string CurrentInstitutionName { get; set; }
        public int BasicDocumentId { get; set; }
        public string RegistrationNumber { get; set; }
        public string RegistrationNumberYear { get; set; }
        public DateTime? RegistrationDate { get; set; }
        public string Note { get; set; }
        public bool IsGranted { get; set; }
        public int? DiplomaId { get; set; }
        public bool ArbitraryCurrentInstitutionName { get; set; }

    }
}
